src/Orleans.Core/Messaging/ClientMessageCenter.cs

```diff
@@ -212,9 +213,23 @@ async Task SendAsync(ValueTask<Connection> task, Message message)
                                 connection.RemoteEndpoint);
                         }
                     }
-                    catch
+                    catch (Exception exception)
                     {
-                        this.SendMessage(message);
+                        if (message.RetryCount < MessagingOptions.DEFAULT_MAX_MESSAGE_SEND_RETRIES)
+                        {
+                            ++message.RetryCount;
+
+                            _ = Task.Factory.StartNew(
+                                state => this.SendMessage((Message)state),
+                                message,
+                                CancellationToken.None,
+                                TaskCreationOptions.DenyChildAttach,
+                                TaskScheduler.Default);
+                        }
+                        else
+                        {
+                            this.RejectMessage(message, $"Unable to send message due to exception {exception}", exception);
+                        }
```

Comment:
The purpose here is to circuit break cases where no gateways are available. If a transient exception occurs, we will retry up to `DEFAULT_MAX_MESSAGE_SEND_RETRIES` times, but promptly reject if the count has been exceeded.